xbmc/interfaces/json-rpc/PlayerOperations.cpp

```diff
@@ -489,6 +490,61 @@ JSONRPC_STATUS CPlayerOperations::Zoom(const std::string &method, ITransportLaye
   }
 }
 
+JSONRPC_STATUS CPlayerOperations::SetViewMode(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  CVariant viewMode = parameterObject["viewmode"];
+  CVariant zoom = parameterObject["zoom"];
+  CVariant pixelRatio = parameterObject["pixelratio"];
+  CVariant verticalShift = parameterObject["verticalshift"];
+  CVariant stretch = parameterObject["nonlinearstretch"];
+
+  // init with current values from settings
+  double newZoom = g_application.GetAppPlayer().GetVideoSettings().m_CustomZoomAmount;
+  double newPixelRatio = g_application.GetAppPlayer().GetVideoSettings().m_CustomPixelRatio;
+  double newVerticalShift = g_application.GetAppPlayer().GetVideoSettings().m_CustomVerticalShift;
+  bool newStretch = g_application.GetAppPlayer().GetVideoSettings().m_CustomNonLinStretch;
+
+  if (zoom.isDouble())
+  {
+    newZoom = zoom.asDouble();
+  }
+
+  if (pixelRatio.isDouble())
+  {
+    newPixelRatio = pixelRatio.asDouble();
+  }
+
+  if (verticalShift.isDouble())
+  {
+    newVerticalShift = verticalShift.asDouble();
+  }
+
+  if (stretch.isBoolean())
+  {
+    newStretch = stretch.asBoolean();
+  }
+
+  if (viewMode.isInteger())
+  {
+    g_application.GetAppPlayer().SetRenderViewMode(static_cast<int>(viewMode.asInteger()),
+                                                   newZoom,
+                                                   newPixelRatio,
+                                                   newVerticalShift,
+                                                   newStretch);
+    return ACK;
+  }
+  return InvalidParams;
+}
+
+JSONRPC_STATUS CPlayerOperations::GetViewMode(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  result["viewmode"] = g_application.GetAppPlayer().GetVideoSettings().m_ViewMode;
+  result["zoom"] = CDisplaySettings::GetInstance().GetZoomAmount();
+  result["pixelratio"] = CDisplaySettings::GetInstance().GetPixelRatio();
+  result["verticalshift"] = CDisplaySettings::GetInstance().GetVerticalShift();
```

Comment:
Can nonlinearstretch boolean be returned too?

Reply:
No it’s internally used but not exposed by settings.

Reply:
How about `CDisplaySettings::GetInstance().IsNonLinearStretched()` ?